### ISSUE TITLE
Weird type signature in `std`

### ISSUE THREAD

**Issue by KAAtheWiseGit** [nushell/nushell#10977]
`run-tests` `--threads` flag has the following type signature:

```
  --threads <CompleterWrapper(Int, 423)>
```

The code for it is:

```Nushell
--threads: int@"nu-complete threads",
```

I think it should be `<Int>`, not a completion wrapper with the closure.

### Configuration

| key                | value                                              |
| ------------------ | -------------------------------------------------- |
| version            | 0.86.1                                             |
| branch             | mkdir                                              |
| commit_hash        | 6b48486f268520a455eea4a302eddd7a5ec1ec63           |
| build_os           | linux-x86_64                                       |
| build_target       | x86_64-alpine-linux-musl                           |
| rust_version       | rustc 1.71.1 (eb26296b5 2023-08-03) (Alpine Linux) |
| cargo_version      | cargo 1.71.1                                       |
| build_time         | 2023-10-26 19:51:06 +03:00                         |
| build_rust_channel | debug                                              |
| allocator          | mimalloc                                           |
| features           | default, sqlite, trash, which, zip                 |
| installed_plugins  |                                                    |

**Comment by amtoine** [comment]
yeah i've seen that already :+1: 

it's not an issue with the standard library per se => it can be reproduced as follows:
```nushell
def foo [] { [1, 2, 3] }
def bar [--x: int@foo] {}
```
```nushell
> help bar
Usage:
  > bar {flags}

Flags:
  --x <CompleterWrapper(Int, 658)> -
  -h, --help - Display the help message for this command

Input/output types:
  #┬input┬output
  0│any  │any
  ─┴─────┴──────
```

**Comment by NotTheDr01ds** [comment]
(Found when searching for something related)

This appears to be fixed.  @amtoine's example now shows:

```nu
  --x: <completable<int>>
```

And `run-tests` now shows:

```nu
  --threads: <completable<int>> - Amount of threads to use for parallel execution. Default: All threads are utilized
```

That's in `std help`.  Built-in help just shows ` --x <int>`

**Comment by KAAtheWiseGit** [comment]
Can confirm.  I think I'll close this issue.  A new one can be opened for `std help` vs `help` inconsistencies

**Comment by amtoine** [comment]
what version of nushell are you using @NotTheDr01ds ?

with Nushell 0.98.0, i get `--x <int> -` with my example :thinking:

**Comment by NotTheDr01ds** [comment]
We're getting the same thing, then.  Looking above, you used to get:

```
--x <CompleterWrapper(Int, 658)> -
```

**Comment by NotTheDr01ds** [comment]
I mean, technically there's still an extra dash hanging out there where the help-comment should be, but the completion closure is gone. I thought that was the original focus of the bug.

**Comment by amtoine** [comment]
ooh my bad, the better help was with std :eyes:

**Comment by NotTheDr01ds** [comment]
No worries - It's a nitpick, but I just fixed it.  PR coming as soon as tests complete.

**Comment by NotTheDr01ds** [comment]
A nitpick on my end, that is - I don't know about whether it is for you or not ;-)

**Comment by NotTheDr01ds** [comment]
Note - Original fix for the closure was in #13531
